Stream_support/include/CGAL/IO/binary_file_io.h

```diff
@@ -20,45 +20,81 @@
 
 #include <CGAL/config.h>
 #include <iostream>
-#include <boost/cstdint.hpp>
+#include <cstdint>
 
 namespace CGAL {
 
 inline void
-I_Binary_write_integer32(std::ostream& out, boost::int32_t i) {
+I_Binary_write_uinteger32(std::ostream& out, std::uint32_t u) {
+    out.write( (char*)(&u), 4);
+}
```

Comment:
Should not the code deal with the endianness? Or maybe we no longer care about endianness, because big-endian platforms are becoming a rarity.

Reply:
That was my assumption (that we can ignore it). I see there are some variants for big endianness in this same file indeed.

This file seems to assume the endianness of the function is checked beforehand, as `I_Binary_read/write_*` are little endian functions while `I_Binary_read/write_big_endian_*` are big endian. I think it would make better sense to use the general name as a wrapper to one or the other depending on the config, but anyway, that is not how it was done until then.

For my own code (in classification), I can write the big endian variants but I can't really test them. Otherwise, I could add an assertion fail if the system is detected as big endian, which would at least notify potential big-endian users that this is not supported (who can then notify us so we work on this support). What do you thiunk?

Reply:
A function like `I_Binary_write_integer32` should swap the bytes if `CGAL_LITTLE_ENDIAN` **is not** defined, and they should probably renamed `I_Binary_write_little_endian_integer32` to explicit the expected endianness of the file.

I am also surprised because I do not see where those functions are used, in CGAL.

Reply:
From what I see, they are only used for the OFF scanner (in the same Stream Support package). I don't know if that will still be the case after the IO feature is integrated.